ui/lib/sync/addon/components/secrets/page/destinations/create-and-edit.ts

```diff
@@ -0,0 +1,71 @@
+/**
+ * Copyright (c) HashiCorp, Inc.
+ * SPDX-License-Identifier: BUSL-1.1
+ */
+
+import Component from '@glimmer/component';
+import { action } from '@ember/object';
+import { tracked } from '@glimmer/tracking';
+import { task } from 'ember-concurrency';
+import { waitFor } from '@ember/test-waiters';
+import { inject as service } from '@ember/service';
+import { findDestination } from 'vault/helpers/sync-destinations';
+import errorMessage from 'vault/utils/error-message';
+
+import type SyncDestinationModel from 'vault/models/sync/destination';
+import { ValidationMap } from 'vault/vault/app-types';
+import type FlashMessageService from 'vault/services/flash-messages';
+import type RouterService from '@ember/routing/router-service';
+import type StoreService from 'vault/services/store';
+
+interface Args {
+  destination: SyncDestinationModel;
+}
+
+export default class DestinationsCreateForm extends Component<Args> {
+  @service declare readonly flashMessages: FlashMessageService;
+  @service declare readonly router: RouterService;
+  @service declare readonly store: StoreService;
+
+  @tracked modelValidations: ValidationMap | null = null;
+  @tracked invalidFormMessage = '';
+  @tracked error = '';
+
+  get displayName() {
+    return findDestination(this.args.destination.type)?.displayName;
+  }
+
+  @task
+  @waitFor
+  *save(event: Event) {
+    event.preventDefault();
+
+    const { destination } = this.args;
+    const { isValid, state, invalidFormMessage } = destination.validate();
+
+    this.modelValidations = isValid ? null : state;
+    this.invalidFormMessage = isValid ? '' : invalidFormMessage;
+    if (!isValid) return;
```

Comment:
Nit picking on some code organization but changing the `if` to look for the truthy value and proceeding with the try catch block feels a bit nicer to me and then there is no need for the blank return.

Reply:
Sure! I find conditional blocks chunky so tend to avoid when possible. But I'll refactor if that feels nicer 😄